jbpm-flow/src/main/java/org/jbpm/process/instance/ProcessRuntimeImpl.java

```diff
@@ -425,13 +425,13 @@ public void matchCreated(MatchCreatedEvent event) {
                     // new activations of the rule associate with a state node
                     // signal process instances of that state node
                     String ruleName = event.getMatch().getRule().getName();
-                    if ( ruleName.startsWith( "RuleFlowStateNode-" )) {
-                        int index = ruleName.indexOf( '-',
-                                                      18 );
-                        index = ruleName.indexOf( '-',
-                                                  index + 1 );
-                        String eventType = ruleName.substring( 0,
-                                                               index );
+                    String rulePrefix = "RuleFlowStateNode-";
+                    if (ruleName.startsWith(rulePrefix)) {
+                        int index = ruleName.indexOf('-', rulePrefix.length() + 1);
+                        int numberOfHyphens = (int) ruleName.chars().filter(ch -> ch == '-').count();
+                        // remove the last 2
+                        index = ordinalIndexOf(ruleName, '-', numberOfHyphens - 1);
+                        String eventType = ruleName.substring(0, index);
```

Comment:
could you please explain what's the intension here?

and is this backward compatible ...

Reply:
@mswiderski  yep. Problem is that you don't know how many hyphens has the process id. But you know how many there are at the end. So you need to remove the last two. and then you get the eventType. It should be backward compatible. 
rule + processId+ uniqueId +....

Reply:
as long as this is backward compatible then I am fine with it